src/io/sampling.jl

```diff
@@ -41,7 +44,17 @@ function adjust_samples(spec::DataFrame, df::DataFrame)::DataFrame
 
     # Same for fogging/shading
     not_fogged = (df.fogging .== 0) .& (df.SRM .== 0)
-    df[not_fogged, contains.(names(df), "shade_")] .= 0.0
+    df[not_fogged, contains.(names(df), "fog_") .& contains.(names(df), "shade_")] .= 0.0
+
+    # Normalize MCDA weights for fogging scenarios
+    is_guided = df.guided .> 0
+    df[(df.fogging .> 0.0) .& is_guided, weights_fog_crit.fieldname] .= mcda_normalize(
+        df[(df.fogging .> 0.0) .& is_guided, weights_fog_crit.fieldname],
+    )
+    # Normalize MCDA weights for seeding scenarios
+    df[.!(not_seeded) .& is_guided, weights_seed_crit.fieldname] .= mcda_normalize(
+        df[.!(not_seeded) .& is_guided, weights_seed_crit.fieldname],
+    )
```

Comment:
Line 46 still needs to be addressed.

I also don't consider the DataFrame selectors to be very readable but if you're okay with it then I'll live.

Reply:
I've changed them again